engine/docker/docker.go

```diff
@@ -324,18 +324,37 @@ func (e *Docker) create(ctx context.Context, pipelineConfig *spec.PipelineConfig
 		)
 	}
 
+	originalImage := step.Image
+	overriddenImage := originalImage
+
+	// override image registry for internal images
+	// this is short term solution
+	// override if no auth is present
+	if isHosted && (step.Auth == nil || step.Auth.Username == "" || step.Auth.Password == "") {
```

Comment:
For internal images didn't we decide to override regardless of auth being present? Is this auth for the default docker connector (or the connector specified in the step)?

Reply:
IMO we should override only for default connector that is used for internal images, for customer overridden connector or customer specified images we need not override the registry

Reply:
Yes if a connector is provided with auth we should keep using it. Only for anonymous auth we need to use GAR